plugins/imagebase/plugin.js

```diff
@@ -6,6 +6,168 @@
 ( function() {
 	'use strict';
 
+	function getFocusedWidget( editor ) {
+		return editor.widgets.focused;
+	}
+
+	function isLinkable( widget ) {
+		return widget && typeof widget.parts.link !== 'undefined';
```

Comment:
I find duck typing for linking feature a bit naive. It's very possible that 3rd parties have `link` part used, in this part these will freak out.

How about providing definite a way to tell that widget implements given feature? For instance it could be simply an array in `widgetDefinition` like `widgetFeatures` with `name`s of features registered.